ambari-server/src/main/java/org/apache/ambari/server/controller/internal/ServiceGroupResourceProvider.java

```diff
@@ -78,6 +80,7 @@
   public static final String SERVICE_GROUP_CLUSTER_NAME_PROPERTY_ID = RESPONSE_KEY + PropertyHelper.EXTERNAL_PATH_SEP + "cluster_name";
   public static final String SERVICE_GROUP_SERVICE_GROUP_ID_PROPERTY_ID = RESPONSE_KEY + PropertyHelper.EXTERNAL_PATH_SEP + "service_group_id";
   public static final String SERVICE_GROUP_SERVICE_GROUP_NAME_PROPERTY_ID = RESPONSE_KEY + PropertyHelper.EXTERNAL_PATH_SEP + "service_group_name";
+  public static final String SERVICE_GROUP_MPACKNAME_PROPERTY_ID = RESPONSE_KEY + PropertyHelper.EXTERNAL_PATH_SEP + "mpacks";
```

Comment:
This is an awkward construct.  Either do the string directly or use PropertyHelper.getPropertyId(...).

Reply:
@ncole This is useful for when we document the API using Swagger annotations.  The compiler does not allow using strings generated by `getPropertyId()` method call in annotation parameters.  Using the string directly in multiple places is prone to typos.  Hence this awkward construct was born.

Anyway, this question does not have much to do with this PR, as @scottduan just followed the existing usage pattern in the lines above.

Reply:
I think mpack:sg is many to many relationship finally. At this time, we just have temporary restriction for 1:1, also I did not validate mpack information. This will be done in the following mpack work.

Reply:
I don't think it should be M2M - it should be M2O. Also, it shouldn't be based on names, but on specific IDs of the mpack. There shouldn't be ambiguity about which MPack a SG is based off of.

Reply:
@adoroszlai - it's a constant, spelling should not be an issue.  You can't use constants in Swagger?  That would be extremely disappointing.

Reply:
@jayush , how do you think? Based on my understand, SG can pick up service from different mpacks, so this should be M2M. 
@jonathan-hurley , because we do not consider version. If we use mpack id, that means we also  consider mpack version. As far as I know, now we only consider mpack name, not including version.

Reply:
A SG should not be able to pickup services from different mpacks - it will break a lot of upgrade flow and assumptions. Additionally, it needs to be associated with a specific mpack in order to provide a way of querying what SG's are compatible with given mpack.

Reply:
mpack provides with service dependence list. So when SG picks up service, theoratically it can choose proper service with the right version (it should be a range). There has a mpack design doc, https://docs.google.com/document/d/13y2T_uN8_nrMsNRuqXLnQAm64Edie0k8_WVNOFvKq0U/edit#heading=h.ykmhjj9wbfa8
We may need to schedule a meeting to clarify how to implement mpack feature in HDP3.

Reply:
@ncole It's a limitation of [annotation "element value"](https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.7.1) in Java, not specific to Swagger.  The value must be a [constant expression](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.28), not just any runtime constant.